app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -133,31 +250,66 @@ public void onPageStop(GeckoView geckoView, boolean success) {
                 }
 
                 @Override
-                public void onSecurityChange(GeckoView geckoView, int status) {
+                public void onSecurityChange(GeckoSession session,
+                                             GeckoSession.ProgressListener.SecurityInformation securityInfo) {
                     // TODO: Split current onPageFinished() callback into two: page finished + security changed
-                    isSecure = status == ProgressListener.STATE_IS_SECURE;
+                    isSecure = securityInfo.isSecure;
                 }
             };
         }
 
-        private NavigationListener createNavigationListener() {
-            return new NavigationListener() {
-                public void onLocationChange(GeckoView view, String url) {
+        private GeckoSession.NavigationListener createNavigationListener() {
```

Comment:
\[PMD\] INFO: The method 'createNavigationListener' has a Cyclomatic Complexity of 8\.